aminator/plugins/cloud/ec2.py

```diff
@@ -306,36 +307,72 @@ def is_stale_attachment(self, dev, prefix):
 
     @registration_retry(tries=3, delay=1, backoff=1)
     def _register_image(self, **ami_metadata):
-        context = self._config.context
-        ami = Image(connection=self._connection)
-        ami.id = self._connection.register_image(**ami_metadata)
-        if ami.id is None:
+        """Register the AMI using boto3/botocore components which supports ENA
+           This is the only use of boto3 in aminator currently"""
+
+        # construct AMI registration payload boto3 style
+        request = {}
+        request['Name'] = ami_metadata.pop('name')
```

Comment:
Any chance the keys you're popping from `ami_metadata` do not exist? You're going to end up throwing a `KeyError` if you pop a non-existent key without providing a default value...